include/amici_solver.h

```diff
@@ -189,6 +231,326 @@ class Solver {
       */
     virtual void turnOffRootFinding() = 0;
 
+    /** sensitivity method
+     * @return method enum
+     */
+    AMICI_sensi_meth getSensitivityMethod() const{
+        return sensi_meth;
+    }
+
+    /**
+     * @brief setSensitivityMethod
+     * @param sensi_meth
+     */
+    void setSensitivityMethod(AMICI_sensi_meth sensi_meth) {
+        this->sensi_meth = sensi_meth;
+    }
+
+    /**
+     * @brief getNewtonMaxSteps
+     * @return
+     */
+    int getNewtonMaxSteps() const {
+        return newton_maxsteps;
+    }
+
+    /**
+     * @brief setNewtonMaxSteps
+     * @param newton_maxsteps
+     */
+    void setNewtonMaxSteps(int newton_maxsteps) {
+        this->newton_maxsteps = newton_maxsteps;
+    }
+
+    /**
+     * @brief getNewtonPreequilibration
+     * @return
+     */
+    bool getNewtonPreequilibration() const {
+        return newton_preeq;
+    }
+
+    /**
+     * @brief setNewtonPreequilibration
+     * @param newton_preeq
+     */
+    void setNewtonPreequilibration(bool newton_preeq) {
+        this->newton_preeq = newton_preeq;
+    }
+
+    /**
+     * @brief getNewtonPreconditioner
+     * @return
+     */
+    int getNewtonPreconditioner() const {
+        return newton_precon;
+    }
+
+    /**
+     * @brief setNewtonPreconditioner
+     * @param newton_precon
+     */
+    // TODO: what are valid options?
+    void setNewtonPreconditioner(int newton_precon) {
+        this->newton_precon = newton_precon;
+    }
+
+    /**
+     * @brief getNewtonMaxLinearSteps
+     * @return
+     */
+    int getNewtonMaxLinearSteps() const {
+        return newton_maxlinsteps;
+    }
+
+    /**
+     * @brief setNewtonMaxLinearSteps
+     * @param newton_maxlinsteps
+     */
+    void setNewtonMaxLinearSteps(int newton_maxlinsteps) {
+        this->newton_maxlinsteps = newton_maxlinsteps;
+    }
+
+    /**
+     * @brief getSensitivityOrder
+     * @return
+     */
+    AMICI_sensi_order getSensitivityOrder() const {
+        return sensi;
+    }
+
+    /**
+     * @brief setSensitivityOrder
+     * @param sensi
+     */
+    void setSensitivityOrder(AMICI_sensi_order sensi) {
+        this->sensi = sensi;
```

Comment:
0 <= sensi <=2?

Reply:
Added one for sensi, but for the remaining setters this will be done at another point.